am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java

```diff
@@ -0,0 +1,143 @@
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.CREATE_PERMISSION;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.OTHER_ROLE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.READ_PERMISSION;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.RESOURCE_TYPE;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ROLE_NAME;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ROOT_ATTRIBUTE;
+import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.SERVICE_NAME;
+
+class GetResourceDefinitionsWithCreatePermissionIntegrationTest extends PreconfiguredIntegrationBaseTest {
+    private static AccessManagementService service = initService(AccessManagementService.class);
+    private static DefaultRoleSetupImportService importerService = initService(DefaultRoleSetupImportService.class);
+
+    private final ResourceDefinition resource = resourceBuilder(RESOURCE_NAME);
+    private final ResourceDefinition resource2 = resourceBuilder(RESOURCE_NAME + "2");
+
+    @BeforeEach
+    void setUp() {
+        importerService.addRole(ROLE_NAME, RoleType.RESOURCE, SecurityClassification.PUBLIC, AccessType.ROLE_BASED);
+        importerService.addRole(
+            OTHER_ROLE_NAME, RoleType.RESOURCE, SecurityClassification.PUBLIC, AccessType.ROLE_BASED);
+        importerService.addResourceDefinition(
+            resource2.getServiceName(), resource2.getResourceType(), resource2.getResourceName());
+    }
+
+    @Test
+    void shouldRetrieveResourceDefinitionWhenRecordExists() {
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(ROOT_ATTRIBUTE, CREATE_PERMISSION));
+
+        List<ResourceDefinition> result =
+            service.getResourceDefinitionsWithCreatePermission(Collections.singleton(ROLE_NAME));
+
+        assertThat(result).containsExactly(ResourceDefinition.builder()
+            .resourceName(RESOURCE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .serviceName(SERVICE_NAME)
+            .build());
+    }
+
+    @Test
+    void shouldRetrieveResourceDefinitionWhenRecordExistsWithMultiplePermissions() {
+        Set<Permission> permissions = ImmutableSet.of(Permission.READ, Permission.CREATE);
+
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(ROOT_ATTRIBUTE, permissions));
+
+        List<ResourceDefinition> result =
+            service.getResourceDefinitionsWithCreatePermission(Collections.singleton(ROLE_NAME));
+
+        assertThat(result).containsExactly(ResourceDefinition.builder()
+            .resourceName(RESOURCE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .serviceName(SERVICE_NAME)
+            .build());
+    }
+
+    @Test
+    void shouldRetrieveResourceDefinitionWhenMultipleRecordsExistWithDifferentAttributes() {
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(
```

Comment:
A `createDefaultPermissionGrant` is bit cryptic. I know that it grants access to same resource definition as `resource` instance but I think it would be better to pass what resource definition we grant access to for readability purpose.